### PR TITLE
newsweek.com home page clone, merge request

### PR DESCRIPTION
**Pull request by 8Bts**
Thanks for checking out my project!

Here is the briefing before reviewing the code:

### BASIC STRUCTURE

3 semantic elements are used in order to build a basic HTML structure:

1.  `<nav>` element for the navigation bar contents.
2.  `<main>` element for the sections between the navigation bar and footer section.
3.  `<footer` element for footer section contents.

### LAYOUT TECHNIQUES

To achieve same layout of the actual page, I used the following layout modules:

- **Floats** for building basic layout of main-contents container, that is between the navbar and in the magazine section, and some inner boxes.

- **Bootstrap Grid System**  for basic containers and inner boxes.

- **CSS Grid** for basic containers and inner boxes.

- **Flex Box** by bootstrap and custom classes, for basic containers and inner boxes.

Please review my project before I merge my branch into master and don't forget to give stars ⭐⭐⭐   👍

### CODE REVIEW

**Review by MarioDena — CHANGES_REQUESTED** [review]
Hello! Great work on your project! 👍  I can tell you pay attention to detail on your work.

However your CSS is not following good practices. 

## Changed required 

- [ ] Please clean up your CSS as there are repeated rules, for example `display: inline-block` is used 15 times.  here is an article for help: https://speckyboy.com/good-bad-css-practices/

## Optional Changes

- [ ]  I recommend replacing the screenshot on your readme file, it's quite long and if someone is interested in for example what technologies you are using your screenshot is on the way of easy navigation.

After making the requested changes please resubmit your project for another review
Cheers!

## Code Review by Mario Enrique Delgado Naranjo

**Comment by 8Bts** [comment]
Thank you for your review @MarioDena. I checked my CSS and cleared it as much as I could. The reason for using the same properties many times is because of elements grouping. Those properties are applied to the element groups that are faster and more clear than using utility classes and applying them on HTML one by one.

Cheers!

**Review by trekab — CHANGES_REQUESTED** [review]
Well done @8Bts, :clap:. You are almost there. I have noticed that there are several scenarios in your CSS where you could have relied on bootstrap classes but didn't. For example when it came to styling margins, borders, padding, use of the display property, use of Flex, and many others. It should be noted that the main point of this project is to remember to use bootstrap classes wherever possible([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/using_bootstrap/bootstrap_classes_requirement.png)) for example [Flex](https://getbootstrap.com/docs/4.3/layout/utilities-for-layout/#flexbox-options). 

__Kindly refine your CSS to use more bootstrap classes and submit another code review request__
_Feel free to contact me on slack for any inquiries about this review_
__Treasure Kabareebe__

File: index.html [hunk 1]
```diff
@@ -0,0 +1,1433 @@
...
+              </span>
+              <span>
+                <a href="#">Vantage</a>
+              </span>
+              <span class="border-right-0">
+                <a href="#">Weather</a>
+              </span>
+            </div>
+            <div class="nav-bottom-right">
+              <span class="search-span">Search</span>
```
Comment (rammazzoti2000) [inline index.html]: I noticed that here you used plain text, this is supposed to be an input please fix this

File: css/style.css [hunk 2]
```diff
@@ -0,0 +1,928 @@
...
+nav {
+  position: sticky;
+  top: -3px;
+  z-index: 2;
+}
+
+.nav-top {
+  background-color: #f72210;
+  height: 50px;
+  align-items: center;
```
Comment (rammazzoti2000) [inline css/style.css]: can be used as bootstrap class
`align-items-center`

File: css/style.css [hunk 3]
```diff
@@ -0,0 +1,928 @@
...
+
+.nav-logo {
+  width: 132px;
+  height: 20px;
+  margin-left: 1em;
+}
+
+.nav-top-right span {
+  color: white;
+  font-weight: 600;
```
Comment (rammazzoti2000) [inline css/style.css]: can be used as bootstrap classes
`font-weight-bold`

File: css/style.css [hunk 4]
```diff
@@ -0,0 +1,928 @@
...
+  font-size: 1.15rem;
+}
+
+.btn-subs {
+  color: var(--prime-color);
+  font-weight: 600;
+  font-size: 1.15rem;
+  background-color: white;
```
Comment (rammazzoti2000) [inline css/style.css]: bootstrap class
`font-weight-bold bg-light`

File: css/style.css [hunk 5]
```diff
@@ -0,0 +1,928 @@
...
+  padding: 0.2em;
+}
+
+.btn-subs i { margin-left: 0.5em; }
+
+.btn-nav-menu {
+  background-color: transparent;
+  color: #fb9a94;
+  font-size: 1.8rem;
+  position: relative;
```
Comment (rammazzoti2000) [inline css/style.css]: bootstrap class
`position-relative bg-transparent`

File: css/style.css [hunk 6]
```diff
@@ -0,0 +1,928 @@
...
+  top: 4px;
+}
+
+.nav-bottom {
+  padding: 1em 0.2em;
+  padding-right: 158px;
+  position: relative;
```
Comment (rammazzoti2000) [inline css/style.css]: bootstrap class
`position-relative`

File: css/style.css [hunk 7]
```diff
@@ -0,0 +1,928 @@
...
+  border-bottom: 2px solid transparent;
+}
+
+.nav-bottom .nav-bottom-links {
+  display: flex;
+  flex-flow: row nowrap;
+  justify-content: center;
+  width: 100%;
```
Comment (rammazzoti2000) [inline css/style.css]: bootstrap classes:
`d-flex flex-row flex-nowrap justify-content-center w-100`

File: css/style.css [hunk 8]
```diff
@@ -0,0 +1,928 @@
...
+  border-left: 12px solid white;
+  border-top: 13px solid var(--prime-color);
+}
+
+/* =========================================== Main Contents Section ====================================== */
+
+.main-contents {
+  position: relative;
+  padding: 0;
+  margin-top: 2px;
```
Comment (rammazzoti2000) [inline css/style.css]: bootstrap classes:
`position-relative p-0 mt-3`

**Review by rammazzoti2000 — CHANGES_REQUESTED** [review]
### Hello @8Bts :raised_hands: your project looks very good, the attention to small details and alignments is impressive, good job :clap: 
Also great job on completing the changes required by the previous TSE's.
There are still some small fixes that are preventing me to allow this project
### Required Changes :construction: :construction_worker: 
- [x] Stickler must be enabled :red_circle: 
- [x] Although you added bootstrap classes to your project is not enough, you can do much better, take a look at the inline comments, those are just a few examples I added, but you can use more, please fix that :pray: :point_right: ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/using_bootstrap/bootstrap_classes_requirement.png) from [Pathwrigth](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59540308/))
- [x] In the navbar you used plain text instead of using the input element, please use a real input, it doesn't have look just like the original but similar, see the inline comment bellow for more clarification
image
- [x] In the 'Subscribe' section on large screens there is a alignment issue, also the left content should be aligned in height with the right content; the image from right content is not displaying please fix that
#### Clone: 
image
#### Original:
image
### <em>Kindly fix the REQUIRED issues and send another code review request.

Cheers and Happy coding! :clap: 

##### Code review by Alexandru Bangau

(Feel free to contact me on slack by the handle of alexandru if you have any additional question)</em>

### :stop_sign:  Status: Changes required.

**Review by rammazzoti2000 — CHANGES_REQUESTED** [review]
#### :stop_sign: Status: Required Changes

File: index.html (hunk 1 above)
Comment (8Bts) [reply index.html:75]: +

Comment (8Bts) [reply index.html]: 🆗

File: css/style.css (hunk 2 above)
Comment (8Bts) [reply css/style.css]: 🆗

File: css/style.css (hunk 3 above)
Comment (8Bts) [reply css/style.css]: 🆗

File: css/style.css (hunk 4 above)
Comment (8Bts) [reply css/style.css]: 🆗

File: css/style.css (hunk 5 above)
Comment (8Bts) [reply css/style.css:73]: Unfortunately, there are no such classes in bootstrap

File: css/style.css (hunk 6 above)
Comment (8Bts) [reply css/style.css:80]: no such class

File: css/style.css (hunk 7 above)
Comment (8Bts) [reply css/style.css]: This one is applied to 11 `<span>` elements.

File: css/style.css (hunk 8 above)
Comment (8Bts) [reply css/style.css]: This selection targets 28 elements.

File: css/style.css (hunk 5 above)
Comment (8Bts) [reply css/style.css]: 🆗

File: css/style.css (hunk 7 above)
Comment (rammazzoti2000) [reply css/style.css]: This is a bootstrap project, no discussion, no compromises.

File: css/style.css (hunk 8 above)
Comment (rammazzoti2000) [reply css/style.css]: This is a bootstrap project, no discussion, no compromises.

**Comment by 8Bts** [comment]
Thanks for review @rammazzoti2000 and helping me to make this project better. I fixed the problems you mentioned.

- [x] Enabled stickler

- [x] Changed `<span>` element to `<input>` element. Even defined transition duration and width change on  `:focus` state.

- [x] Fixed bugs on the subscribe section.

- [x] Added even more bootstrap classes instead of using CSS. Even removed declarations for grouped elements and added them on HTML one by one. 

### For the next code reviewers 
Please don't ask me to use more bootstrap classes again. This is the maximum usage for my design. Some bootstrap classes can't be responsive and I needed to define custom stylings inside medium and large size breakpoint media queries.
I know that my CSS file looks too big for the Bootstrap based project, but if you check the live version, you can see that I tried my best for making a page that looks very similar to the actual one. In order to achieve this, I needed more CSS code lines.

**Comment by lekegitrepo** [comment]
#### Hi @8Bts :+1: 
#### You did a good job by implementing the changes required in the last review. This project is approved.

#### Note: As mentioned by the previous reviewer; try to make the height of the left and right contents in the subscribe section to be the same.

### Reviewed by Temitayo Adeleke.

- Github: [@lekegitrepo](https://github.com/lekegitrepo)
- Twitter: [@leketemi](https://twitter.com/leketemi)
- Linkedin: [linkedin](https://www.linkedin.com/in/temitayo-adeleke/)

**Review by lekegitrepo — APPROVED** [review]
#### Status: Approved.

**Pull request merged**